app/controllers/teachers_controller.rb

```diff
@@ -260,4 +280,33 @@ def sanitize_params
   def load_pages
     @pages ||= Page.where(viewer_permissions: Page.viewable_pages(current_user))
   end
+
+  def update_primary_email(primary_email)
+    return unless primary_email.present?
+
+    # First, ensure the current primary email is marked as not primary if it's not the same as the new one
+    @teacher.email_addresses.where(primary: true).each do |email_record|
```

Comment:
This seems like we should be able to write a `@teacher.primary_email` call since there should only be 1 primary email for each teacher. Not sure why we need a loop here.

Reply:
My existing `@teacher.primary_email` is a method call to only get a email address value itself:
```ruby
  def primary_email
    # ||:email this code is temporary for this PR: https://github.com/cs169/BJC-Teacher-Tracker-App/pull/49
    # to make sure at least original data in db still work and passed the existing tests
    self[:email] || email_addresses.find_by(primary: true)&.email
  end
```

To update the `primary` attribute for old email address record, based on your advice, I feel it could be
```ruby
teacher.email_addresses.find_by(primary: true)&.update(primary: false)
```